src/pages/Profile.js

```diff
@@ -154,6 +155,114 @@ const NihLink = ({ nihToken }) => {
 }
 
 
+const FenceLink = ({ provider, displayName }) => {
+  const decodeProvider = state => state ? JSON.parse(atob(state)).provider : ''
+
+  const extractToken = (provider, { state, code }) => {
+    const extractedProvider = decodeProvider(state)
+    return extractedProvider && provider === extractedProvider ? code : undefined
+  }
+
+  const queryParams = qs.parse(window.location.search, { ignoreQueryPrefix: true })
+  const token = extractToken(provider, queryParams)
+  const redirectUrl = `${window.location.origin}/${Nav.getLink('fence-callback')}`
+
+  /*
+   * Hooks
+   */
+  const [{ username, issued_at: issuedAt }, setStatus] = useState({})
+  const [href, setHref] = useState(undefined)
+  const [isLoadingStatus, setIsLoadingStatus] = useState(false)
+  const [isLoadingAuthUrl, setIsLoadingAuthUrl] = useState(false)
+  const [isLinking, setIsLinking] = useState(false)
+  const signal = useCancellation()
+
+  const { User } = Ajax(signal)
+
+  Utils.useOnMount(() => {
+    const linkFenceAccount = _.flow(
+      withErrorReporting('Error linking NIH account'),
+      Utils.withBusyState(setIsLinking)
+    )(async () => {
+      setStatus(await User.linkFenceAccount(provider, token, redirectUrl))
+    })
+
+    if (token) {
+      const profileLink = `/${Nav.getLink('profile')}`
+      window.history.replaceState({}, '', profileLink)
+      linkFenceAccount()
+    }
+  })
+
+  Utils.useOnMount(() => {
```

Comment:
TOL: We haven't settled on a strong convention for organizing on-mount code. A few thoughts/ideas for consideration:
* The bulk of the code (in this case the `loadFenceStatus` function) could be moved outside the hook, which would reduce the hook body itself to a line or two.
* In the case of independent code paths, is it clearer to have multiple `useOnMount` hooks, or a single one that invokes all the paths?

Reply:
We chatted about this in person and I agreed that pulling the functions out made sense. Once I did that, I had very little reason for having multiple calls to `useOnMount` so I'm consolidating those.